datastore/longhorn.go

```diff
@@ -1715,11 +1765,19 @@ func (s *DataStore) CheckEngineImageReadiness(image string, nodes ...string) (is
 	return true, nil
 }
 
-// CheckEngineImageReadyOnAtLeastOneVolumeReplica checks if the IMAGE is deployed on the NODEID and on at least one of the the volume's replicas
-func (s *DataStore) CheckEngineImageReadyOnAtLeastOneVolumeReplica(image, volumeName, nodeID string, dataLocality longhorn.DataLocality) (bool, error) {
-	isReady, err := s.CheckEngineImageReadiness(image, nodeID)
+func (s *DataStore) CheckImageReadiness(image string, backendStoreDriver longhorn.BackendStoreDriverType, nodes ...string) (isReady bool, err error) {
+	if backendStoreDriver == longhorn.BackendStoreDriverTypeV2 {
+		// TODO: check readiness of instance manager image
```

Comment:
Have a ticket for this

Reply:
Will remove the `TODO` and revisit it when adding upgrade codes.